pkgs/native_assets_cli/lib/src/model/pipeline_config.dart

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) 2024, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:pub_semver/pub_semver.dart';
+
+abstract class PipelineConfigImpl {
```

Comment:
These are just getters. Consider actually moving the shared implementation in the subclasses in to here.

(Altenatively, make this just an interface and implement it in the subclasses instead of extending.)